_posts/2017-04-05-solita-java-trainee.md

```diff
@@ -0,0 +1,72 @@
+
+At the beginning of the first day the other trainees and I were welcomed to Solita with Starter packs containing all kinds of Solita goodies. We were given a refreshment presentation about the program in general and the agenda for the day as well as the coming weeks. A tour of the premises helped us to get our bearings in the Tampere office.
+
+![Day One](/img/solita-java-trainee/dayone.png)
+
+Soon it was time to go our separate ways to meet our home bases for the rest of the program. The assigned teams had been selected based on a compatibility evaluation of our backgrounds and existing skills, and the needs of the projects. So it was that I met the project team and was assigned a mentor and a seat. All set!
+
+## Practice Assignment
+
+New employees at Solita start with a practice assignment, and though technically not an employee, I was no exception. A modified version of the regular Solita practice assignment was what I would concentrate on for the first few weeks. Should I have any issues or questions on how to continue, my mentor and other teammates would be there to assist.
+
+The assignment was about implementing a full-stack CRUD web discussion forum application. For my part, the emphasis was on the back-end, as Java programming was what would be of most use in the project, whereas front-end technologies would not be so important at this time.
+
+So, most of my time was spent on fleshing out the bare-bones template implementation of a Java Spring Boot application with a REST API and controller, service, and DAO layers, plus some alternative ways of implementation, particularly to replace the DAOs. I also received suggestions, as well as hints on how to go about it, on adding support for exception handling and tests. Obviously the front-end needed a some attention as well, to make the forum usable. Most of the work there I carried out with AngularJS 1.5, with a few small touches of HTML and CSS.
+
+## Trainings and Self-studies
+
+The Java Trainee program included 15 days of technical classroom training. The trainings were front-loaded to the first half of the program, which was a good thing for getting up to speed with Java. The courses, carried out by Tieturi, consisted of _Web Standards_, _JavaScript Development_, _Java Development Basics_, _Java Fundamentals_, _Java Web Development_, and _Java Web Services Development_, in that order.
+
+Even though I had used Java before in my career in one project, it had been long enough since then that I felt all the courses were useful to me. Even the basics was a welcome refreshment. Thinking of the practice assignment, it would have been nice to be able to attend the last couple of courses earlier, as the material covered a lot of what the assignment was about. In a perfect world, all the courses could be packed a little closer together and perhaps be conducted still a little earlier in the program. All in all, though, I think the content and the timing of the trainings was still very good.
```

Comment:
Even the basics _were_ a welcome
the content and the timing of the trainings _were_ still very good